docs/designers-developers/developers/block-api/block-attributes.md

```diff
@@ -2,13 +2,18 @@
 
 ## Common Sources
 
-Attribute sources are used to define the strategy by which block attribute values are extracted from saved post content. They provide a mechanism to map from the saved markup to a JavaScript representation of a block.
+Attribute sources are used to define how the block attribute values are extracted from saved post content. They provide a mechanism to map from the saved markup to a JavaScript representation of a block.
 
 If no attribute source is specified, the attribute will be saved to (and read from) the block's [comment delimiter](/docs/designers-developers/key-concepts.md#delimiters-and-parsing-expression-grammar).
 
-Each source accepts an optional selector as the first argument. If a selector is specified, the source behavior will be run against the corresponding element(s) contained within the block. Otherwise it will be run against the block's root node.
+The keys specified in the attributes source object are named as you see fit. The result of the attribute source definition will be assigned as a value to each key.
+
+If no selector argument is specified, the source definition runs against the block's root node. If a selector argument is specified, it will run against the specified element(s) contained within the block.
+
+The selector specified can be an HTML tag, but also anything queryable such as a class or id attribute, see examples below.
+
+Under the hood, attribute sources are a superset of functionality provided by [hpq](https://github.com/aduth/hpq), a small library used to parse and query HTML markup into an object shape. 
```

Comment:
Maybe "superset of functionality provided" to "superset of the functionality provided"?

Reply:
:+1:  I like all the suggestions, thanks!